### PR TITLE
cpu/stm32_common: added LPTIM based rtt implementation

### PR DESCRIPTION
**Pull request by haukepetersen**
This rtt implementation works for all STM CPUs, that have a low power timer (LPTIM), so far mainly the STM32L0 and STM32L4.

So far it is only configured for the STM32L4. Support for the STM32L0 is still open (follow-up PR), as for this to work the clock init for the L0 needs some love (i.e. system wide initialization of LSE/LSI.

### CODE REVIEW

File: boards/nucleo-l476/include/periph_conf.h [hunk 1]
```diff
@@ -234,6 +234,17 @@ static const spi_conf_t spi_config[] = {
 #define DAC_NUMOF           (0)
 /** @} */
 
+/**
+ * @brief   RTT configuration
+ *
+ * On the STM32Lx platforms, we always utilize the LPTIM1.
```
Comment (aabadie) [inline boards/nucleo-l476/include/periph_conf.h]: s/utilise/use

File: cpu/stm32_common/periph/rtt_lptim.c [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+void rtt_poweron(void)
+{
+#ifdef RCC_APB1ENR1_LPTIM1EN
+    periph_clk_en(APB1, RCC_APB1ENR1_LPTIM1EN);
+#else
+    periph_clk_en(APB1, RCC_APB1ENR_LPTIM1EN);
+#endif
+}
+
+
```
Comment (aabadie) [inline cpu/stm32_common/periph/rtt_lptim.c]: extraline

File: boards/nucleo-l476/include/periph_conf.h [hunk 3]
```diff
@@ -234,6 +234,17 @@ static const spi_conf_t spi_config[] = {
 #define DAC_NUMOF           (0)
 /** @} */
 
+/**
+ * @brief   RTT configuration
```
Comment (aabadie) [inline boards/nucleo-l476/include/periph_conf.h]: should be `@name`

File: cpu/stm32l4/include/cpu_conf.h [hunk 4]
```diff
@@ -36,7 +36,7 @@ extern "C" {
  * @{
  */
 #define CPU_DEFAULT_IRQ_PRIO            (1U)
-#define CPU_IRQ_NUMOF                   (82U)
+#define CPU_IRQ_NUMOF                   (81U)
```
Comment (aabadie) [inline cpu/stm32l4/include/cpu_conf.h]: 82 is correct, le L432 has 82 interrupts (I started to look at nucleo32-l432 already). The last in CRS global interrupt.

**Comment by aabadie** [comment]
Appart a few inline comments, I tested and this works very well.

File: boards/nucleo-l476/include/periph_conf.h (hunk 1 above)
Comment (haukepetersen) [inline boards/nucleo-l476/include/periph_conf.h]: I prefer utilize

**Comment by haukepetersen** [comment]
addressed comments

File: cpu/stm32l4/include/cpu_conf.h (hunk 4 above)
Comment (aabadie) [inline cpu/stm32l4/include/cpu_conf.h]: you forgot this one

File: boards/nucleo-l476/include/periph_conf.h (hunk 1 above)
Comment (aabadie) [inline boards/nucleo-l476/include/periph_conf.h]: Maybe rephrase then: `On the STM32Lx platforms, LPTIM1 is always used` ?

This makes me think that we could probably makes this configurable in the board periph_conf file ? as the stm32l476rg has 2 LPTIM.

File: cpu/stm32l4/include/cpu_conf.h (hunk 4 above)
Comment (haukepetersen) [inline cpu/stm32l4/include/cpu_conf.h]: Hm, the `stm32l476` has definitely only 81, so the CRS must be specific to the nucleo-l432?! I'd say we go with 81 for this PR and you just add an ifdef case for the `l432` when you add that CPU. Sounds good?

Comment (aabadie) [inline cpu/stm32l4/include/cpu_conf.h]: I would keep the 82 and add this extra line in vectors.c:
```
(void*) (0UL)
```

Comment (haukepetersen) [inline cpu/stm32l4/include/cpu_conf.h]: but that would waste 4byte of flash...

Comment (aabadie) [inline cpu/stm32l4/include/cpu_conf.h]: Not sure that it makes a big difference on a CPU with 1MB of flash but I accept the argument. I'll update this with preprocessor macros in #6838.

**Comment by haukepetersen** [comment]
squashed

File: cpu/stm32_common/periph/rtt_lptim.c [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "irq.h"
+#include "periph/rtt.h"
+
+/* this driver is only valid for STM CPUs that provide LPTIMERs */
+#ifdef LPTIM1
```
Comment (aabadie) [inline cpu/stm32_common/periph/rtt_lptim.c]: Maybe changing this line to:
```c
#if (defined(LPTIM1) && defined(RTT_NUMOF) && RTT_NUMOF)
```
will fix the CI until the L0 boards are correctly configured.

**Comment by haukepetersen** [comment]
fixed guard for `rtt_lptim`

**Comment by aabadie** [comment]
All green. Go !
